audio/csound/Portfile

```diff
@@ -68,6 +69,21 @@ if {${configure.build_arch} ni [list arm i386 ppc]} {
     default_variants-append +double
 }
 
+if {${configure.build_arch} in [list ppc ppc64]} {
+    # Absence of altivec is not automatically detected.
+    if {[catch {sysctl hw.vectorunit} result] || $result == 0} {
+        configure.cppflags-append \
+                        -DPFFFT_SIMD_DISABLE
+
+        # As this machine is a G3, force a local build so we don't download a buildbot-built
+        # version possibly built on a G4+ or Rosetta.
+        archive_sites
+    } else {
+        # We do not want to add -mcpu= here, that is not safe.
+        configure.optflags-append -mtune=native
+    }
+}
+
```

Comment:
This would seem to build altivec support depending on what the build architecture is rather than the architecture running the port?

Reply:
Doing this right is a bit tricky.

To do it properly, the best way is to have support for altivec pathways discovered at runtime and used it available. Some very well-written software does that, but it's too hard for most folks to figure out.

The next best thing is to force a build of the software on the machine that will be running it by clearing archive_sites, and then testing for the presence of altivec on the build machine by doing a system check. You can see this done in the ffmpeg port:

```
platform powerpc {
    # absence of altivec is not automatically detected
    if {[catch {sysctl hw.vectorunit} result] || $result == 0} {
        configure.args-append \
                    --disable-altivec

        # as this machine is a G3, force a local build so we don't download a buildbot-built
        # version possibly built on a G4+ or cross-compiled from an Intel system
        archive_sites
    }
}}}
```

The ffmpeg port assumes that most people will have a PPC machine that can run Altivec, so that is the default build, built by the buildbot (if there is one). But if the test shows the machine does not have altivec when you go to do the installation, the portfile disables altivec and clears archive_sites, so you have to build it locally (without altivec).

Reply:
In this particular spot in the portfile, the build just sets "mtune=native" however. That makes gcc deliver an optimal build for the machine it is currently building on. In that case, you would have to clear archive_sites always, so that you would always build the software on the machine it will be running on.

Reply:
@kencu 
> In that case, you would have to clear archive_sites always

Thank you, I will do this.